src/odd.rs

```diff
@@ -168,9 +168,9 @@ impl<'de, T: Deserialize<'de> + Integer + Zero> Deserialize<'de> for Odd<T> {
         D: Deserializer<'de>,
     {
         let value: T = T::deserialize(deserializer)?;
-
-        if bool::from(value.is_odd()) {
-            Ok(Self(value))
+        let odd = Self::new(value);
```

Comment:
If you do:

```suggestion
        let odd = Option::<Self>::from(Self::new(value));
```

...you can handle the rest using `Option`/`Result` combinators